src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -225,8 +225,11 @@ private Task ProcessGetId(HttpContext context)
                     format,
                     endOfMessage: true);
 
-                await state.Application.Output.WriteAsync(message);
-
+                // REVIEW: Do we want to return a specific status code here if the connection has ended?
+                if (await state.Application.Output.WaitToWriteAsync())
```

Comment:
If you know the channel is unbounded, then you also know that writes will always succeed (standard caveats of OOMs and the like) unless the channel has been closed. Thus, as long as the channel hasn't been closed, TryWrite will aways return true, and WaitToWriteAsync will always complete synchronously with a true result. This would be documented, and is inherent to the nature of it being unbounded.
If you don't know whether the channel has room, e.g. it's bounded, then you would need to do something like a loop with WaitToWriteAsync/TryWrite or use WriteAsync to wait until there's room to write.